lib/widgets/lightbox.dart

```diff
@@ -208,11 +213,244 @@ class _LightboxPageState extends State<_LightboxPage> {
   }
 }
 
+class VideoLightboxPage extends StatefulWidget {
+  const VideoLightboxPage({
+    super.key,
+    required this.routeEntranceAnimation,
+    required this.message,
+    required this.src,
+  });
+
+  final Animation routeEntranceAnimation;
+  final Message message;
+  final Uri src;
+
+  @override
+  State<VideoLightboxPage> createState() => _VideoLightboxPageState();
+}
+
+class _VideoLightboxPageState extends State<VideoLightboxPage> {
+  // TODO(#38): Animate entrance/exit of header and footer
+  bool _headerFooterVisible = false;
+
+  VideoPlayerController? _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.routeEntranceAnimation.addStatusListener(_handleRouteEntranceAnimationStatusChange);
+    // We delay initialization by a single frame to make sure the
+    // BuildContext is a valid context, as its needed to retrieve
+    // the PerAccountStore & ZulipLocalizations during initialization.
+    SchedulerBinding.instance.addPostFrameCallback((_) => _initialize());
+  }
+
+  Future<void> _initialize() async {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    assert(debugLog('VideoPlayerController.networkUrl(${widget.src})'));
+    _controller = VideoPlayerController.networkUrl(widget.src, httpHeaders: {
+      if (widget.src.origin == store.account.realmUrl.origin) ...authHeader(
+        email: store.account.email,
+        apiKey: store.account.apiKey,
+      ),
+      ...userAgentHeader()
+    });
+    _controller!.addListener(_handleVideoControllerUpdates);
+
+    try {
+      await _controller!.initialize();
+      await _controller!.play();
+    } catch (error) { // TODO(log)
+      assert(debugLog("VideoPlayerController.initialize failed: $error"));
+      if (mounted) {
+        await showDialog(
+          context: context,
+          barrierDismissible: false,
+          builder: (BuildContext context) => AlertDialog(
+            title: const Text('Unable to play video'), // TODO(i18n)
+            actions: [
+              TextButton(
+                onPressed: () => Navigator.popUntil(context, (route) => route is MaterialAccountPageRoute),
+                child: dialogActionText(zulipLocalizations.errorDialogContinue)),
+            ]));
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    _controller?.removeListener(_handleVideoControllerUpdates);
+    _controller?.dispose();
+    widget.routeEntranceAnimation.removeStatusListener(_handleRouteEntranceAnimationStatusChange);
+    super.dispose();
```

Comment:
Cool, good to see the order of these updated to match (in reverse order) changes to the order of the corresponding initializations.